### ISSUE TITLE
getspec fails with old Avantes files

### ISSUE THREAD

**Issue by celiason** [rmaia/pavo#89]
Some older Avantes (ttt) files have "0804016U1- " on first line. This code `raw <- raw[grepl('^[0-9\\-]', raw)]` interprets it as reflectance data. I think it will work to put `raw[grepl(paste0("^\\d.*?\\", decimal), raw)]` instead?

**Comment by Bisaloo** [comment]
@celiason, I don't think that's the issue. This line gets removed later in the code ([L162](https://github.com/rmaia/pavo/blob/master/R/getspec.R#L162)).

If you want more proof this is not the cause of your issue, you can have a look at the [`avantes_export.ttt` file](https://github.com/rmaia/pavo/blob/master/inst/testdata/avantes_export.ttt) which is used in the [tests](https://github.com/rmaia/pavo/blob/master/tests/testthat/test-import.R#L7-L8).

Would you mind sharing the file you're having issues with? It should help finding what's wrong. If you don't want to share it, your best bet if probably to compare it closely to  [`avantes_export.ttt` file](https://github.com/rmaia/pavo/blob/master/inst/testdata/avantes_export.ttt) and find out how they differ.

You are right this regexp could be improved though. I don't like relying on an error, which is basically what happens here. R fails to convert `0804016U1-` to a numeric and outputs `NA`. This `NA` is then dropped later during the processing.

**Comment by Bisaloo** [comment]
@rmaia, do you remember why you changed to this instead of removing all lines with text in https://github.com/rmaia/pavo/commit/a81bb8456bea1ad31063626a73466a04c8fb6772?

For future reference if I work on `getspec`, the reason why `(E|e)` was not taken into account: https://github.com/rmaia/pavo/commit/94bd046afa5b5084af1c260fda6c2c7f9e1f38d9

**Comment by celiason** [comment]
OK, so the first line of the file has `0804016U1- O=5 E=10`. When I remove everything after '=' it works. The "fix" I mentioned above works fine on my machine (but now I'm not sure why). Are the `=` maybe messing up the regexps?

Here's the top of the offending ttt file:
```
0804016U1- O=5 E=10
Integration time: 100.00 ms
Average: 10 scans
Nr of pixels used for smoothing: 0
Data measured with spectrometer name: 0804016U1
Wave   ;Transmittance
[nm]   ;[% transmittance]
300.00;2.0931
301.00;3.0184
302.00;2.9916
```

**Comment by Bisaloo** [comment]
Because of the 2 spaces in your first line, `getspec` builds a 3 columns dataframe (`wl`, `transmittance/reflectance`, `wl`), which then fails when we take the first and last column on [L168](https://github.com/rmaia/pavo/blob/master/R/getspec.R#L168).

**Comment by rmaia** [comment]
The reason I had to change it was that some new version of OO spectrometers were producing files that had letters in the data columns - not just `Ee`, but other text - and that was causing the function to cut off that from the data. I'm trying to find the files but haven't yet. I agree that relying on an error is kinda awful, the issue is these spectrometer output files are very inconsistent and it is a sisyphean task to make rules that work for all of them. So I had to prioritize function over elegance - which admittedly can end up being the reason the boulder rolls down again in the future...
